src/PriorityList/LegacyImplementation.php

```diff
@@ -225,9 +230,7 @@ public function key()
      */
     public function next()
     {
-        $node = next($this->items);
-
-        return $node ? $node['data'] : false;
+        next($this->items);
```

Comment:
The fact that this no longer return smells like BC break

Reply:
Yes and no.

`Iterator::next()` is documented as being `return void`, and the return value is ignored; it determines if the call to `next()` produces a value by calling `valid()` - and that method has this logic (and always did).

Basically, we were breaking the contract established by the interface previously, and the return value from this was never intended to be used. I consider this a programming mistake instead. (And I likely produced it myself...)

Reply:
I can do one of the following:

- Revert the change. This will require (a) documenting the return value (we didn't before, and instead inherited from the `Iterator` interface... so it was _documented_ as `void`), and (b) marking the return value as deprecated. The question is then what to do about the PHP 8.1 implementation, which _must_ be `void`; we'll have different behavior between the two versions at that point. Additionally, I'll likely need to suppress some additional Psalm errors, due to the difference in the signatures between the interface and implementation.
- Keep the change. The question will be whether or not we call it out, as, again, the previously documented return value was `void`, which would have implied that the return value did not matter.

Knowing that information, what is your vote?